pkg/topology/pslice/pslice_test.go

```diff
@@ -422,22 +421,102 @@ func chkNotExists(t *testing.T, ps *pslice.PSlice, addrs ...swarm.Address) {
 	}
 }
 
+var (
+	base = test.RandomAddress()
+	bins = int(swarm.MaxBins)
+)
+
 func BenchmarkAdd(b *testing.B) {
-	var (
-		base = test.RandomAddress()
-		ps   = pslice.New(16, base)
-	)
+	ps := pslice.New(bins, base)
 
-	for i := 0; i < 16; i++ {
-		for j := 0; j < 300; j++ {
+	for i := 0; i < bins; i++ {
+		for j := 0; j < b.N; j++ {
```

Comment:
This is not correct way to write benchmarks. The benchmark iteration is under another loop.